common/workunit/workunit.cpp

```diff
@@ -14496,11 +14497,22 @@ void executeThorGraph(const char * graphName, IConstWorkUnit &workunit, const IP
         }
     }
 
+    // NB: check for expected success state (WUStateWait). If any other state, abort.
     {
         Owned<IWorkUnit> w = &workunit.lock();
         WUState state = w->getState();
-        if (WUStateFailed == state)
-            throw makeStringException(0, "Workunit failed");
+        if (WUStateWait != state) // expected state from successful Thor run from above
+        {
+            switch (state)
+            {
+                case WUStateAborting:
+                    throw new WorkflowException(0, "Workunit abort requested", 0, WorkflowException::ABORT, MSGAUD_user);
+                case WUStateFailed:
+                    throw makeStringException(0, "Workunit failed");
+                defualt:
```

Comment:
typo: will not compile!

Reply:
that's nasty - it does compile! It's seeing it as a goto statement label :( - you'd hope it would spot this likely bad pattern and error, but nope.

Reply:
Now fixed (squashed in)

Reply:
That is horrible!